activerecord/test/cases/strict_loading_test.rb

```diff
@@ -24,6 +24,9 @@ def test_strict_loading!
     assert_raises ActiveRecord::StrictLoadingViolationError do
       developer.audit_logs.to_a
     end
+
+    developer.strict_loading!(false)
+    assert_not_predicate developer, :strict_loading?
```

Comment:
Can you add an assertion that `developer.audit_logs` doesn't raise? I think we should also have a test showing that it also works if the strict loading is defined in the relation. See `strict_loading_audit_logs` below.

Reply:
Done in https://github.com/rails/rails/pull/41181/commits/8223023591a75390a512e813bc0f56398db225c1